### PR TITLE
Valid Knight move

### PR DESCRIPTION
**Pull request by chris-campbell**
Missed names branch as Bishop vs Knight, so ignore that. Completed knight valid move and added test to make sure it works. Also added a gem called 'awesome print', it helps making objects being viewed in the rails c more readable. Just add 'ap' in front of commands and it prettifys them.

### CODE REVIEW

File: app/models/knight.rb [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+    (x_diff) == 2 && (y_diff == 1) ? true : false
+    (y_diff) == 2 && (x_diff == 1) ? true : false
+    
+  end
+  
+  def determine_piece_color(x, y)
+    if !present?(x, y)
+      game.pieces.find_by(position_x: x, position_y: y).color
+    end
+  end
```
Comment (ulalu) [inline app/models/knight.rb]: Oh man, lovely work on implementing obstructed. I'm going to crib this for kings in the next day or two.

File: app/models/piece.rb [hunk 2]
```diff
@@ -97,5 +114,5 @@ def obstructed?(x, y)
     end
   end
   
-end # end of class
+end
```
Comment (ulalu) [inline app/models/piece.rb]: Thanks so much for commenting this whole thing out. It was definitely readable before, but this is so much simpler. :+1:

Comment (chris-campbell) [reply app/models/piece.rb]: Thanks! I never feel fully done,  so if anything is clear i'm always here.

File: app/models/knight.rb (hunk 1 above)
Comment (chris-campbell) [reply app/models/knight.rb]: Thanks a lot! Thats sound good to me. I think I should move the determine_piece_color to Piece.rb

**Pull request merged**